### PR TITLE
Fix AD Claim Mapping to User Extended model

### PR DESCRIPTION
**Pull request by WCONFR**
Fix for reported issue: https://github.com/snok/django-auth-adfs/issues/241

### CODE REVIEW

File: django_auth_adfs/backend.py
Comment (tim-schilling) [inline django_auth_adfs/backend.py:177]: You should use `isinstance(claim, dict)` so that it'll support other types of dict-like objects.

File: django_auth_adfs/backend.py
Comment (tim-schilling) [inline django_auth_adfs/backend.py:194]: The value at `mapped_fields[field]` is already available as `claim`, use that.

File: django_auth_adfs/backend.py
Comment (tim-schilling) [inline django_auth_adfs/backend.py:192]: This check needs to happen in the outer loop to prevent the `else` branch below from causing an error in `getattr(user, field)` when it doesn't exist.

File: django_auth_adfs/backend.py
Comment (tim-schilling) [inline django_auth_adfs/backend.py:165]: What do you think about `field_claim_map` or `claim_mapping` rather than `mapped_fields`? I think it should be closer to the setting name.

**Review by tim-schilling — CHANGES_REQUESTED** [review]
Thank you for opening this PR! I think this would help others as well.

The two changes that I would require to see is the usage of `isinstance` and reworking the check if the field doesn't exist. The other comments are suggestions/comments.

This PR does needs tests and documentation.

File: django_auth_adfs/backend.py
Comment (WCONFR) [reply django_auth_adfs/backend.py]: Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a

File: django_auth_adfs/backend.py
Comment (WCONFR) [reply django_auth_adfs/backend.py]: Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a

File: django_auth_adfs/backend.py
Comment (WCONFR) [reply django_auth_adfs/backend.py]: Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a

File: django_auth_adfs/backend.py
Comment (WCONFR) [reply django_auth_adfs/backend.py]: Done: https://github.com/snok/django-auth-adfs/pull/242/commits/91f7ee5273cacda41e7a790e378bd8698fc3f39a

**Comment by WCONFR** [comment]
Thank you for the great feedback ! I implemented all the requested changes.

However, not really sure what you require for the Documentation and Testing (very first PR on a public repo), could you help me out?

Thanks in advance !

**Comment by WCONFR** [comment]
@tim-schilling Added commit https://github.com/snok/django-auth-adfs/pull/242/commits/ac6d4984f4cf66bfddfcef87194cbd122e4d5c5b which holds documentation of the PR

**Comment by tim-schilling** [comment]
Regarding the test here's one way to solve it:

You'll need to create a models file in the `tests/` directory that creates a user extension model. You could name it profile, and set it to have a `OneToOneField` to User. Give it some random property.

Then the actual test should be similar to `test_boolean_claim_mapping`, but using the CLAIM_MAPPING setting that will try to update the Profile model for the user.

File: tests/test_authentication.py
Comment (tim-schilling) [inline tests/test_authentication.py:273]: This seems to be in the wrong direction. The assertion on `user.profile.employee_id` should work.

**Comment by tim-schilling** [comment]
Ah, shoot. There's a problem here. Django doesn't automatically create related fields. So in the case of the User instance being created, you would need to wire up a `post_safe` signal receiver or something equivalent to create the profile when the user is created. That would guarantee that the profile instance would exist when we hit the `update_user_attribute` flow.

**Comment by tim-schilling** [comment]
@WCONFR alright this should be good. If any of the build actions fail besides the coverage case, they need to be resolved.

**Comment by WCONFR** [comment]
Ok, fully understood. Thanks for your work on this, I will be patiently waiting for the second review.

**Comment by JonasKs** [comment]
This looks good to me. 

> Ideally we'd have better coverage of the exception flows in update_user_attributes

Yeah, ideally 😁 I'm happy with merging this as it is, though. 😊

**Review by JonasKs — APPROVED** [review]

**Comment by WCONFR** [comment]
@JonasKs Done, thanks for completing the merge

**Comment by JonasKs** [comment]
Thank you so much for the contribution, and to @tim-schilling for helping out like a champ! 😊

**Comment by WCONFR** [comment]
@JonasKs Do I need to do something to merge the PR since it has been approved ? Don't want to rush you, just want to be sure that you're not waiting on action from my end.

**Comment by JonasKs** [comment]
No, you're good, thank you. I've just been packed with work so haven't got around to it. I'll do it later 😊

**Comment by JonasKs** [comment]
Sorry for the delay! This has been released in `1.10.0`. Thank you! 

https://github.com/snok/django-auth-adfs/releases/tag/1.10.0

**Pull request merged**
